app/src/org/commcare/dalvik/activities/EntityMapActivity.java

```diff
@@ -1,288 +1,186 @@
+        GoogleMap.OnInfoWindowClickListener {
     private static final String TAG = EntityMapActivity.class.getSimpleName();
-    private final static int LOCATION_PERMISSIONS_REQUEST = 1;
-
-    private MapView map;
-    private MyLocationOverlay mMyLocationOverlay;
-    private Geocoder mGeoCoder;
-    private LocationManager mLocationManager;
-    
-    private CommCareSession session;
+    private static final int MAP_PADDING = 50;  // Number of pixels to pad bounding region of markers
 
-    private EntityOverlay mEntityOverlay;
-    private Vector<Entity<TreeReference>> entities;
-    
-    @Override
-    protected void onCreate(Bundle icicle) {
-        super.onCreate(icicle);
-        String licenseKey = getLicenseKey();
-        map = new MapView(this, licenseKey);
-        
-        this.setContentView(map);
-        
-        mGeoCoder = new Geocoder(this);
+    private final CommCareSession session = CommCareApplication._().getCurrentSession();
+    private final SessionDatum selectDatum = session.getNeededDatum();
 
-        session = CommCareApplication._().getCurrentSession();
-        Vector<Entry> entries = session.getEntriesForCommand(session.getCommand());
+    private final Vector<Pair<Entity<TreeReference>, LatLng>> entityLocations = new Vector<>();
+    private final HashMap<Marker, TreeReference> markerReferences = new HashMap<>();
 
-        SessionDatum selectDatum = session.getNeededDatum();
-        Detail detail = session.getDetail(selectDatum.getShortDetail());
-        NodeEntityFactory factory = new NodeEntityFactory(detail, this.getEC());
-        
-        Vector<TreeReference> references = getEC().expandReference(selectDatum.getNodeset());
-        
-        entities = new Vector<>();
-        for(TreeReference ref : references) {
-            entities.add(factory.getEntity(ref));
-        }
+    private GoogleMap mMap;
 
-        Log.d(TAG, "Entities generated");
-        
-        map.displayZoomControls(true);
-        mMyLocationOverlay = new MyLocationOverlay(this, map);
-        mMyLocationOverlay.runOnFirstFix(new Runnable() { public void run() {
-            map.getController().animateTo(mMyLocationOverlay.getMyLocation());
-        }});
-        
-        Drawable defaultMarker = this.getResources().getDrawable(R.drawable.marker);
-        mEntityOverlay = new EntityOverlay(defaultMarker, map) {
+    @Override
+    @TargetApi(11)
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.entity_map_view);
 
-            @Override
-            protected void selected(TreeReference ref) {
-                Intent i = new Intent(EntityMapActivity.this.getIntent());
-                SerializationUtil.serializeToIntent(i, EntityDetailActivity.CONTEXT_REFERENCE, ref);
-                
-                setResult(RESULT_OK, i);
+        MapFragment mapFragment = (MapFragment) getFragmentManager()

```

Comment:
`getFragmentManager` should probably be `getSupportFragmentManager` to be compat with other devices